addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -575,3 +614,293 @@ def drop_off_picked(self, continue_batch, move_line_ids, location_barcode, resul
         #     self.close()
 
         return self
+        
+    def get_single_batch(self, user_id=None):
+        """
+        Search for a picking batch in progress for the specified user.
+        If no user is specified, the current user is considered.
+
+        Raise a ValidationError in case it cannot perform a search
+        or if multiple batches are found for the specified user.
+        """
+        PickingBatch = self.env["stock.picking.batch"]
+
+        user_id = self._check_user_id(user_id)
+        batches = PickingBatch.search(
+            [("user_id", "=", user_id), ("state", "=", "in_progress")]
+        )
+        batch = None
+
+        if batches:
+            if len(batches) > 1:
+                raise ValidationError(
+                    _("Found %d batches for the user, please contact " "administrator.")
+                    % len(batches)
+                )
+
+            batch = batches
+
+        return batch
```

Comment:
Nit: Some of this logic can be tidied up e.g. if the search returns no results it will be an empty recordset. Perhaps something like this:
```
batches = PickingBatch.search(BLAH)

if len(batches) > 1:
    raise ValidationError(BLAH)

return batches or None
```